common/persistence/dataInterfaces.go

```diff
@@ -551,20 +558,20 @@ type (
 
 	// VersionHistoryItem contains the event id and the associated version
 	VersionHistoryItem struct {
-		EventID int64
-		Version int64
+		eventID int64
+		version int64
 	}
 
 	// VersionHistory provides operations on version history
 	VersionHistory struct {
-		BranchToken []byte
-		History     []VersionHistoryItem
+		branchToken []byte
+		items       []*VersionHistoryItem
```

Comment:
nit: I like the renaming but are there reasons to change this from slice of items to slice of pointer to items ? Given that each historyItem is two int64s, it is very cache friendly to have these as slice of items instead of pointers.

Reply:
was think about the same reason when using slice of structs, i.e. cache friendly
the reason to change to pointer is, currently correctness is more important than performance, and the focus is to make the logic works end to end.

using a slice of struct maybe good for perf, but now (during the development), perf is not a major concern and i prefer correctness (less bug).

after the initial dev, i may end up changing this back

Reply:
#2053